internal/resolver/resolver.go

```diff
@@ -139,6 +139,20 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 				return nil, errors.Wrapf(FeatureGateOCI.Error(),
 					"repository %s is an OCI registry", d.Repository)
 			}
+
+			// This works, fake it to test mocking the version
+			// tags := []string{"0.1.0", "0.1.1", "0.2.0"}
+			tags := d.OCITagVersions
```

Comment:
The tags here aren't sorted. I didn't test it but reading through the oras tag code it doesn't appear to be. This list is turned into `ChartVersion` which are still unsorted by version. But, the version constraint handling expects it to be sorted. So, these tags need to be sorted and non-semver tags pulled out as part of the process. This will be especially important because there is work to support sigstore which uses tags.

You can find a sorting example at https://github.com/Masterminds/semver/blob/master/collection_test.go (tests as an example)

Reply:
Thanks this is helpful. Will do soon.

Reply:
Been collabing w Andy, who just pushed a fix for this - added directly to registry `(*Client).tags` 💯